src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetries.cs

```diff
@@ -29,9 +40,10 @@ protected internal override void Setup(FeatureConfigurationContext context)
             var retryPolicy = GetRetryPolicy(context.Settings);
 
             context.Container.RegisterSingleton(typeof(SecondLevelRetryPolicy), retryPolicy);
+
             context.Pipeline.Register<SecondLevelRetriesBehavior.Registration>();
 
-            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress()), DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress(), b.Build<FailureInfoStorage>()), DependencyLifecycle.InstancePerCall);
```

Comment:
I think that we should create the `FailureStorageInfo` by hand. We want to get rid of builder and currently we depend on `MoveFaultsToErrorQueue` behavior to create it.